### PR TITLE
logging creation of tables

### PR DESCRIPTION
**Pull request by ReevesAk**
Logging to the console the when tables are created
Fixes 127#
Screenshot from 2019-08-01 22-00-41

### CODE REVIEW

**Comment by Sirmorrison** [comment]
> 
> 
> Logging to the console the when tables are created
> Fixes 127#
> Screenshot from 2019-08-01 22-00-41

Your screen shot shows multiple instances of `creating table` logged for each source added to the table created. 
You should have only one instance of `Creating table` logged for each table created.

File: exchanges/hubs.go [hunk 1]
```diff
@@ -200,6 +200,7 @@ func (hub *TickHub) Run(ctx context.Context) {
 
 	registerStarter()
 	hub.CollectAll(ctx)
+	log.Info("Creating Exchange table")
```
Comment (ademuanthony) [inline exchanges/hubs.go]: This is not where the table is been created

File: mempool/mempool.go [hunk 2]
```diff
@@ -221,6 +221,7 @@ func (c *Collector) StartMonitoring(ctx context.Context) {
 		mempoolDto.Revocations = len(revocations)
 
 		err = c.dataStore.StoreMempool(ctx, mempoolDto)
+		log.Info("Creating Mempool table")
```
Comment (ademuanthony) [inline mempool/mempool.go]: !

File: pow/collectors.go [hunk 3]
```diff
@@ -140,6 +140,7 @@ func (pc *Collector) Collect(ctx context.Context) {
 				log.Error(err)
 			}
 			err = pc.store.AddPowData(ctx, data)
+			log.Info("Creating poW table")
```
Comment (ademuanthony) [inline pow/collectors.go]: !

File: vsp/vsp.go [hunk 4]
```diff
@@ -128,6 +128,7 @@ func (vsp *Collector) collectAndStore(ctx context.Context) error {
 	}
 
 	numberStored, errs := vsp.dataStore.StoreVSPs(ctx, *resp)
+	log.Info("Creating VSP table")
```
Comment (ademuanthony) [inline vsp/vsp.go]: !

**Comment by ReevesAk** [comment]
I believe I have. 
The behavior raedah requested for was for a message to printed when tables are created for each module. Tables are created only once, which is the first time you load from postgres db. The message should be seen if/when you delete the day from the old database and creating a new one. 
Have a different opinion?

**Comment by ReevesAk** [comment]
Implementing the review will not print the message

**Comment by ademuanthony** [comment]
> I believe I have.
> The behavior raedah requested for, was for a message to be printed when tables are created for each module. Tables are created only once, which is the first time you load from postgres db. The message should be seen if/when you delete the day from the old database and creating a new one.
> Have a different opinion?

The main point in the review is; '`In order to have a well organised log, let us move all table creation logic to one place and terminate execution if we receive any error during the process.`'

**Comment by ReevesAk** [comment]
Screenshot from 2019-08-06 00-52-58

File: main.go [hunk 5]
```diff
@@ -184,105 +187,33 @@ func _main(ctx context.Context) error {
...
-			}
-		}
-
 		go collector.StartMonitoring(ctx)
-	}else {
-		fmt.Println("Creating memepool table...")
 	}
 
 	if !cfg.DisableVSP {
-		if exists := db.VSPInfoTableExits(); !exists {
```
Comment (Sirmorrison) [inline main.go]: this method is missing in your `createTablesAndIndex()` function.

**Comment by Sirmorrison** [comment]
image

resolved.

**Pull request merged**